packages/synapse-react-client/src/synapse-queries/QueryBatchUtils.ts

```diff
@@ -0,0 +1,27 @@
+import { chunk } from 'lodash-es'
+import { UseQueryOptions } from '@tanstack/react-query'
+
+/**
+ * Given a list of items, batch them into groups of maxBatchSize and create a QueryOptions object for each chunk based on the provided mapToQueryOptionsFn
+ * @param listToBatch
+ * @param maxBatchSize
+ * @param mapToQueryOptionsFn
+ */
+export function batchQueries<
```

Comment:
At this layer, the caller must handle the complexity of having multiple result objects.

If we move it to the SynapseClient layer, it probably makes more sense if SynapseClient merges the result objects (or returns an array of results that the caller then has to merge?).

Reply:
Another consideration is it would be nice to keep logic out of our handwritten SynapseClient so we can simply replace it with client code generated from the OpenAPI definition.